File: set_value.ts
// Random TypeScript function generated on 2025-08-18

interface ProcessRequest {
    id: number;
    name: string;
    value: number;
    active: boolean;
}

function check_config(config_omd: ProcessRequest[]): ProcessRequest[] {
    const multiplier: number = 2;
    
    return config_omd
        .filter((item: ProcessRequest) => item.active)
        .map((item: ProcessRequest, index: number) => ({
            ...item,
            value: item.value * multiplier
        }));
}

// Example usage
const testData: ProcessRequest[] = [
    { id: 1, name: "Item1", value: 100, active: true },
    { id: 2, name: "Item2", value: 47, active: false }
];

const result = check_config(testData);
console.log('Processed items:', result);

export { check_config, ProcessRequest };

File: set_value.cpp
// Random C++ function generated on 2025-08-21
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

struct Item {
    int id;
    std::string name;
    double value;
    
    Item(int id, const std::string& name, double value) 
        : id(id), name(name), value(value) {}
};

std::vector<Item> update_value(const std::vector<Item>& items) {
    const double multiplier = 3.0;
    std::vector<Item> result;
    
    std::transform(items.begin(), items.end(), std::back_inserter(result),
        [multiplier](const Item& item) {
            return Item(item.id, item.name + "_processed", item.value * multiplier);
        });
    
    return result;
}

int main() {
    std::vector<Item> testData = {
        Item(1, "item1", 18.0),
        Item(2, "item2", 29.0)
    };
    
    auto result = update_value(testData);
    std::cout << "Processed " << result.size() << " items" << std::endl;
    
    return 0;
}
